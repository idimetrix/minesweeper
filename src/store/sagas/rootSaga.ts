import { takeLatest } from 'redux-saga/effects';

import { watchOnGame, handleCreateGame } from '../../game/gameActions';
import { actions } from '../../game/gameReducers';

export function* watcherSaga() {
  yield takeLatest(actions.initialiazeGame.type, watchOnGame);
  yield takeLatest(actions.createGame.type, handleCreateGame);
}
