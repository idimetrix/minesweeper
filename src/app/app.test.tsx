// eslint-disable-next-line import/no-unresolved
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';
import App from './app';
import { watcherSaga } from '../store/sagas/rootSaga';

const sagaMiddleware = createSagaMiddleware();
const mockStore = configureStore([sagaMiddleware]);

describe('<App />', () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('renders correctly and to match snapshot', () => {
    const initialState = {
      game: {
        map: [],
        message: '',
      },
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Start button appears in first render 1', () => {
    const initialState = {
      game: {
        map: [],
        message: '',
      },
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(wrapper.text().includes('Minesweeper')).toBe(true);
    expect(wrapper.text().includes('Press START and start playing Minesweeper!')).toBe(true);
    expect(wrapper.text().includes('Start')).toBe(true);
  });

  it('Start button appears in first render 2', () => {
    const initialState = {
      game: {
        map: [
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
          '□□□□□□□□□□',
        ],
        message: '',
      },
    };
    const store = mockStore(initialState);

    sagaMiddleware.run(watcherSaga);

    const wrapper = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(wrapper.getByText('Play again')).toBeTruthy();
  });
});
