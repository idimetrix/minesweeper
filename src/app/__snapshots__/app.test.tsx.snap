// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly and to match snapshot 1`] = `
<div
  className="makeStyles-layout-2"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 makeStyles-container-3 css-p7cgzs-MuiPaper-root"
  >
    <div
      className="makeStyles-header-4"
    >
      <p
        className="makeStyles-headText-1"
      >
        Minesweeper
      </p>
      <p
        className="makeStyles-message-10"
      >
        
      </p>
    </div>
    <div
      className="makeStyles-content-5 makeStyles-smallContent-6"
    >
      <div
        className="ant-result ant-result-info"
      >
        <div
          className="ant-result-icon"
        >
          <span
            aria-label="table"
            className="anticon anticon-table"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="table"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 208H676V232h212v136zm0 224H676V432h212v160zM412 432h200v160H412V432zm200-64H412V232h200v136zm-476 64h212v160H136V432zm0-200h212v136H136V232zm0 424h212v136H136V656zm276 0h200v136H412V656zm476 136H676V656h212v136z"
              />
            </svg>
          </span>
        </div>
        <div
          className="ant-result-title"
        >
          Press START and start playing Minesweeper!
        </div>
      </div>
    </div>
    <div
      className="makeStyles-footer-7"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth makeStyles-levelSelector-9 css-q8hpuo-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          id="level-select-label"
        >
          Level
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="level-select-label level-select"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="level-select"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            1
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="1"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1in441m"
            >
              <span>
                Level
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-startButton-8 css-ke5b6m-MuiButtonBase-root-MuiButton-root"
        data-testid="start-game-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
